Source/Scene/WebMapServiceImageryProvider.js

```diff
@@ -371,6 +419,35 @@ define([
             set : function(enablePickFeatures)  {
                 this._tileProvider.enablePickFeatures = enablePickFeatures;
             }
+        },
+
+        /**
+         * Gets or sets a clock that is used to get keep the time used for time dynamic parameters.
+         * @memberof WebMapServiceImageryProvider.prototype
+         * @type {Clock}
+         */
+        clock : {
+            get : function() {
+                return this._timeDynamicImagery.clock;
```

Comment:
I know this is already merged, but there's a bug here.  If `times` isn't passed in the constructor, `this._timeDynamicImagery` is undefined so this (and all other such references in the getters/setters below) will throw.  Once a WMSImageryProvider is initialized as "non-dynamic" it's impossible to make it dynamic, so any references to dynamic properties should return undefined.